docs: Updating documentation structure and content in the tutorial section

AnaisUrlichs opened the pull request:
## Description
- Converting the Quick-start section into tutorials
- Moving integrations into the tutorials section
- adding three tutorials; Trivy Kubernetes, GitOps, and installing Trivy into ovh cloud
- I made the installation section more visible by moving it to the main index.md

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/aquasecurity/trivy?pullRequest=2333) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/aquasecurity/trivy?pullRequest=2333) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: AnaisUrlichs<br/>:x: Anais Urlichs<br/><hr/>**Anais Urlichs** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/aquasecurity/trivy?pullRequest=2333) it.</sub>

AnaisUrlichs commented:
@itaysk @knqyf263 could you please check the PR again?

AnaisUrlichs commented on docs/tutorials/overview.md:
```diff
@@ -0,0 +1,17 @@
+**Tip:** Make sure that your tutorial is concise about a specific use case or integration. 
+
+How to add a tutorial
+1. Simply create a new `.md` file in the [tutorials](tutorials) folder of the docs.
```
Ah yes, good point, will include that

AnaisUrlichs commented on docs/tutorials/other-resources/references.md:
```diff
@@ -1,6 +1,19 @@
+# Further Reading
+
+## Presentations
+- Aqua Security YouTube Channel
```
no, will rewrite it -- thanks for pointing it out

AnaisUrlichs commented:
@knqyf263 could you please review and merge the docs with the next release or similar?

knqyf263 commented:
I don't have a strong opinion. I'm ok if Itay approves it. One thing to note is that we added some updates on our doc such as [SBOM attestation](https://aquasecurity.github.io/trivy/v0.30.3/docs/attestation/sbom/). We need to make sure all the changes are reflected to this PR.

AnaisUrlichs commented:
@knqyf263 I've merged the main branch into this one in one of my last commits

AnaisUrlichs commented:
@knqyf263 just merged main into it

knqyf263 commented:
@AnaisUrlichs Thanks.
@itaysk Would you review this PR?

itaysk commented on docs/tutorials/kubernetes/ovhcloud.md:
```diff
@@ -0,0 +1,452 @@
+# Scan for vulnerabilities and misconfigurations of your OVHcloud Managed Kubernetes with Trivy
```
Are we sure this is a "first party" tutorial and not "third party"? do we want to maintain it?

itaysk commented on mkdocs.yml:
```diff
@@ -7,13 +7,30 @@ repo_url: https://github.com/aquasecurity/trivy
+      - Additional Resources: 
+          - Tools: tutorials/other-resources/tools.md
+          - References: tutorials/other-resources/references.md
+          - CKS Reference: tutorials/other-resources/cks.md
```
I cannot comment on the file so commenting here: we could add some context there about what is CKS and why it's in the trivy docs.

AnaisUrlichs replied:
Yes, will add that

itaysk commented on docs/tutorials/kubernetes/ovhcloud.md:
```diff
@@ -0,0 +1,452 @@
+# Scan for vulnerabilities and misconfigurations of your OVHcloud Managed Kubernetes with Trivy
```
Are we sure this is a "first party" tutorial and not "third party"? do we want to maintain it?

knqyf263 replied:
No, we can't. The first party tutorial should be

- GitHub Actions
  - We have an official action
- CircleCI
- Travis CI
- GitLab CI
  - GitLab folks maintain our tutorial
- Bitbucket Pipelines
  - We have [an official repo](https://github.com/aquasecurity/trivy-pipe)
- (AWS CodePipeline)
  - It should be moved to third party as the blog is written by AWS, not us.
- AWS Security Hub
  - as we maintain [asff.tpl](https://aquasecurity.github.io/trivy/v0.30.4/docs/integrations/aws-security-hub/)

AnaisUrlichs replied:
Hmm in addition to those we need tutorials such as Deploy the `Trivy Operator through GitOps` Where do you see those?

itaysk commented on docs/tutorials/overview.md:
```diff
@@ -0,0 +1,24 @@
+
+How to add a tutorial
+
+1. Simply create a new `.md` file in the tutorials folder of the docs
```
is there a specific place for "community tutorials" or just anywhere in tutorials? is this what the "tools" are for? if so, need to refine this instruction and also rename tools. otherwise lets clarify what tools are for

AnaisUrlichs replied:
I would only accept tutorials here that we want to maintain e.g. installing Trivy Operator through Flux, Visualising Metrics in Grafana etc. or similar, re Trivy -- using Trivy in an air-gapped environment -- and similar -- otherwise, tutorials should be added in the additional resource section and just reference an external link. This is how I had it in mind

AnaisUrlichs commented:
@itaysk @knqyf263 I made several changes based on your comments
- removed the Kubernetes tutorial from the tutorials section and added it to the community tutorials
- made it clearer what community tutorials are and where they go in the docs
- clarified what CKS stands for with a link
- merged in latest changes from main

knqyf263 commented on docs/tutorials/integrations/gitops.md:
```diff
@@ -0,0 +1,125 @@
+# Installing the Trivy-Operator through GitOps
```
"CI/CD" section details how to run Trivy in CI/CD, but this section is how to install Trivy Operator through GitOps. I feel like this section is different from others.
Also, do we want to Trivy Operator things in the Trivy doc?

AnaisUrlichs replied:
This boils down to the question -- if Trivy is marketed as the all in one security scanner, do we want to send people do one docs where they can discover all features or to different docs -- or moreover, at what point do we send people to different docs?
When Itay and I spoke, we agreed to put the tutorials first into the trivy docs and then at some point move them over to the operator docs when there are too many -- but this might be counterintuitive since links might not work like expected.

Independently, I could add the GitOps tutorial to the Kubernetes tutorial section

knqyf263 replied:
>When Itay and I spoke, we agreed to put the tutorials first into the trivy docs and then at some point move them over to the operator docs when there are too many -- but this might be counterintuitive since links might not work like expected.

Does it mean we will not have the same tutorial in the Trivy Operator doc? I just want to avoid duplication so that it could reduce the maintenance cost.

itaysk replied:
I think of this more as "installing Trivy in k8s through GitOps" rather than "installing Trivy Operator through GitOps". In my mind Trivy Operator is just another face of Trivy. Does that make sense?

knqyf263 replied:
I'm ok with "installing Trivy in k8s through GitOps". I just wondered if we want to include all Trivy-related tools in the tutorial. For example, Docker Extension is also another face of Trivy. In that case, don't we have the same tutorial in the Trivy Operator doc? Where is the single truth of source?

itaysk replied:
Last time we discussed it we said that:

1. reference docs will live in it's own repo, probably as plain readme files
2. we will link from the menu of trivy docs to the individual readmes/docs or other projects
3. tutorials will live in Trivy docs under dedicated tutorials sections covering different scenarios 

Examples:

1. Trivy git hub actions document with all possible options is in the readme of the action's repo. It is linked to from the integrations section of trivy docs. the tutorial for using trivy in github actions is in trivy docs and linked in the triby tutorials section.
2. trivy semaphore document with all possible options is in the semaphore documentation. It is linked to from the integrations section of trivy docs. the tutorial for using trivy with semaphore is in the semaphore blog and linked in the trivy tutorials section.

does this still makes sense?

AnaisUrlichs replied:
We planned to put a link to Trivy Operator at the top header -- this turned now into Trivy Ecosystem linking to the external repos/docs -- so you are saying that you would like to keep it as Trivy Operator in the header and link to external tools in the integration section?
We could still have the Trivy Ecosystem overview somewhere -- maybe in an overview in integrations?

itaysk replied:
No I didn't mean to revert anything, just wanted to answer Teppei's comment. I think (?) this model I described still works with ecosystem page

knqyf263 replied:
OK, we can keep Operator tutorials then.

knqyf263 commented on mkdocs.yml:
```diff
@@ -7,13 +7,29 @@ repo_url: https://github.com/aquasecurity/trivy
+      - Installation: home/installation.md
+  - Tutorials:
+      - Overview: tutorials/overview.md
+      - Trivy Kubernetes: 
```
nit: IMO, we don't need `Trivy` everywhere like `Trivy AWS` and `Trivy CircleCI` 😆  

```suggestion
      - Kubernetes: 
```

knqyf263 replied:
Not fixed yet?

knqyf263 commented on mkdocs.yml:
```diff
@@ -7,13 +7,29 @@ repo_url: https://github.com/aquasecurity/trivy
+  - Tutorials:
+      - Overview: tutorials/overview.md
+      - Trivy Kubernetes: 
+          - Trivy K8s: tutorials/kubernetes/trivy-kubernetes.md
```
`Trivy K8s` is not clear. I think we should mention what users can do with this tutorial, e.g. "Cluster scanning" or something like that.

itaysk replied:
maybe Kubernetes scanning

knqyf263 replied:
ditto

knqyf263 replied:
ditto

knqyf263 commented on docs/tutorials/other-resources/cks.md:
```diff
@@ -1,16 +1,18 @@
 
-Aqua Security Blog posts
+### Aqua Security Blog posts
 
 - Supply chain security best [practices][supply-chain-best-practices]
 - Supply chain [attacks][supply-chain-attacks]
-- 
+
```
They are not CKS relevant, no?

AnaisUrlichs replied:
yes but they are also not trivy-focused and teach people something

knqyf263 replied:
So, should we move these links to somewhere else? Otherwise, it looks like these links help users pass the CKS exam.

knqyf263 commented on mkdocs.yml:
```diff
@@ -7,13 +7,29 @@ repo_url: https://github.com/aquasecurity/trivy
+          - GitOps: tutorials/integrations/gitops.md       
+      - Additional Resources: 
+          - Tools: tutorials/other-resources/tools.md
+          - Community Tutorials: tutorials/other-resources/references.md
```
That page contains much of our content, so it is not community tutorials.

```suggestion
          - Additional References: tutorials/other-resources/references.md
```

AnaisUrlichs replied:
you have a point

knqyf263 commented on docs/tutorials/other-resources/references.md:
```diff
@@ -0,0 +1,61 @@
+# Additional References
+Below is a list of additional resources from the community.
+
+# Further Reading
```
There are two `#`. We need to remove this line or remove the first line.

AnaisUrlichs replied:
removed this line :)

knqyf263 commented on docs/tutorials/other-resources/references.md:
```diff
@@ -0,0 +1,61 @@
+# Additional References
+Below is a list of additional resources from the community.
```
Most of them are our blogs, not from the community.

AnaisUrlichs replied:
yes, will add more community ones -- you have a point tho hahah

knqyf263 replied:
We may want to have two sections:
- Additional References
   - All of the contents are from Aqua (official)
- Community Tutorials
  - All of the contents are from the community (unofficial)

AnaisUrlichs replied:
I renamed it to additional resources -- the thing is that official vs unofficial is going to be a bit scarce, and people would have to look at multiple places that have the same goal

knqyf263 reviewed (commented):
I said I was ok, but after all I left some comments😄

AnaisUrlichs commented on mkdocs.yml:
```diff
@@ -119,15 +126,13 @@ nav:
-      - References: community/references.md
-      - CKS Reference: community/cks.md
-      - Credits: community/credit.md
+  - Trivy Operator: docs/kubernetes/operator/index.md
```
hmmm Itay and I discussed this -- but good point -- maybe we could put it at the top level
Integrations -- and then reference a page that links to either the docs or the different repo e.g. the Lens Trivy Repo 
People can then easily go through it and find the integrations/tools that work for them. 
Do you prefer to name it `Tools`?

knqyf263 replied:
`Integrations` and `Tools` both work for me.

AnaisUrlichs commented on mkdocs.yml:
```diff
@@ -7,13 +7,29 @@ repo_url: https://github.com/aquasecurity/trivy
-  - Docs:
+  - Getting Started: 
+      - Overview: index.md
+      - Quick Start: home/quickstart.md
+      - Installation: home/installation.md
```
I don't have any strong feelings about the order & agree with you on the dir name 😄 will change it

AnaisUrlichs commented on docs/home/installation.md:
```diff
@@ -248,6 +248,38 @@ podAnnotations: {}
 
 > **Tip**: List all releases using `helm list`.
 
+# Trivy Operator Installation
```
I would add the Docker Extension as a tutorial and then link here to it since it does not seem too straightforward and is more about the use rather than the installation. What do you think?

AnaisUrlichs replied:
actually made the link to Trivy Operator part of the Helm installation. That makes more sense to me (didn't change much)

knqyf263 replied:
>I would add the Docker Extension as a tutorial and then link here to it since it does not seem too straightforward and is more about the use rather than the installation. What do you think?

Hmm. I thought Trivy Operator, Docker Extensions, and other tools would move to `Integrations` or `Tools`. Do we have links in the `Installation` section to all the tools?

itaysk commented:
consider adding tekton tutorial: https://www.redhat.com/architect/cicd-pipeline-openshift-tekton

knqyf263 commented on mkdocs.yml:
```diff
@@ -7,13 +7,29 @@ repo_url: https://github.com/aquasecurity/trivy
+          - AWS CodePipeline: tutorials/integrations/aws-codepipeline.md
+          - AWS Security Hub: tutorials/integrations/aws-security-hub.md
+      - Additional Resources: 
+          - Tools: tutorials/other-resources/tools.md
```
We moved `Tools` to `Ecosystem`. We don't need it anymore, right?

AnaisUrlichs replied:
Removed it

knqyf263 commented on docs/getting-started/installation.md:
```diff
@@ -248,6 +251,44 @@ podAnnotations: {}
+- Scan public and private images that are either locally or in the Docker Hub.
+- Generate an SBOM for your container images.
+
+Watch the [tutorial for further information.](https://youtu.be/99tiVZLMujs)
```
As for `Trivy Operator Installation` and `Docker Desktop Extension`, can we just put "Other tools" and say `See the `Ecosystem` tab?

AnaisUrlichs replied:
Done

knqyf263 commented on mkdocs.yml:
```diff
@@ -7,13 +7,29 @@ repo_url: https://github.com/aquasecurity/trivy
+      - Trivy Kubernetes: 
+          - Trivy K8s: tutorials/kubernetes/trivy-kubernetes.md
+          - GitOps: tutorials/kubernetes/gitops.md       
+      - CI/CD:
```
Can we put `CI/CD` on the `Kubernetes` section? CI/CD integration is the first citizen in Trivy.

AnaisUrlichs replied:
What do you mean?

AnaisUrlichs replied:
ah got it

knqyf263 commented on docs/tutorials/other-resources/tools.md:
```diff
@@ -0,0 +1,96 @@
+| [Trivy Vulnerability Explorer][explorer] | Explore trivy vulnerability reports in your browser and create .trivyignore files interactively. Can be integrated in your CI/CD tooling with deep links.   |
+
+
+## Official Trivy Tools
```
People might try unofficial tools first because community tools are listed first. Can we put official tools before community tools?

AnaisUrlichs replied:
Done :)

knqyf263 reviewed (commented):
I left some small comments. I believe this is the last review.

AnaisUrlichs commented:
@knqyf263 could you please have another look 🙃

knqyf263 commented:
@AnaisUrlichs Would you resolve conflicts and merge the main branch? Then I hope we can merge this PR🤞

AnaisUrlichs commented:
@knqyf263 merged main and resolved conflicts

knqyf263 commented:
I saw several rendering errors and fixed them. Also, I added some small changes.

knqyf263 commented:
@AnaisUrlichs Do you have two accounts? Looks like one of them has not signed CLA yet.
https://github.com/aquasecurity/trivy/pull/2333#issuecomment-1156868203
